pytorch_to_returnn/converter/converter.py

```diff
@@ -156,6 +157,24 @@ def _make_tf_feed_dict(self, input: Data):
     else:
       for i, size in input.size_placeholder.items():
         d[size] = [self._inputs_np.shape[input.get_batch_axis(i)]] * n_batch  # not so relevant
+    if out_entry:
+      def _get_non_deterministic_tensor_entries(entry_: TensorEntry):
+        """
+        Go recursively through all tensor entries to find non deterministic ones
+        """
+        if not all(call_.module.module.is_deterministic for call_ in entry_.output_from_calls):
+          return [entry_]
+        non_det_mods = []
+        for call_ in entry_.output_from_calls:
+          for prev_entry_ in [pe for pe in call_.inputs_args if isinstance(pe, TensorEntry)]:
+            if prev_entry_ != entry_:
+              non_det_mods += _get_non_deterministic_tensor_entries(prev_entry_)
```

Comment:
You probably visit many entries multiple times, and thus exponentially many times. To avoid this, you need to keep track a set of visited entries.